targets/unity-v2/Testing/Tests/Entity/EntityApiTests.cs

```diff
@@ -219,38 +224,44 @@ void OnGetFilesInfo(GetFilesResponse result)
                 GetActualFile(testContext, fileMetaData);
             }
         }
-        void GetActualFile(UUnitTestContext testContext, GetFileMetadata fileData)
+        void GetActualFile(UUnitTestContext testContext, DataModels.GetFileMetadata fileData)
         {
-            PlayFabHttp.SimpleGetCall(fileData.DownloadUrl, PlayFabUUnitUtils.SimpleApiActionWrapper<byte[]>(testContext, TestFileContent),
+            PlayFabHttp.SimpleGetCall(fileData.DownloadUrl,
+                PlayFabUUnitUtils.SimpleApiActionWrapper<byte[]>(testContext, TestFileContent),
                 error =>
                 {
                     testContext.Fail(error);
                 });
         }
         void UploadFile(UUnitTestContext testContext, string fileName)
         {
-            var request = new InitiateFileUploadsRequest
+            var request = new DataModels.InitiateFileUploadsRequest
             {
-                Entity = _entityKey,
+                Entity = new DataModels.EntityKey { Id = _entityId, TypeString = _entityTypeString },
                 FileNames = new List<string>
                 {
                     fileName
                 },
             };
 
-            PlayFabEntityAPI.InitiateFileUploads(request, PlayFabUUnitUtils.ApiActionWrapper<InitiateFileUploadsResponse>(testContext, OnInitFileUpload), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, OnInitFailed), testContext);
+            PlayFabDataAPI.InitiateFileUploads(request, PlayFabUUnitUtils.ApiActionWrapper<DataModels.InitiateFileUploadsResponse>(testContext, OnInitFileUpload), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, OnInitFailed), testContext);
         }
-        void DeleteFiles(UUnitTestContext testContext, List<string> fileName)
+        void DeleteFiles(UUnitTestContext testContext, List<string> fileName, bool shouldEndTest, UUnitFinishState finishState, string finishMessage)
         {
-            var request = new DeleteFilesRequest
+            if (!_shouldDeleteFiles) // Only delete the file if it was created
```

Comment:
Will it gracefully end test after return in all cases?

Reply:
Nope!
This is certainly something we should address in the TestSuite itself, which is that SetUp and TearDown should be able to be long-running async as well.
For now, the "standard case" is that the test won't have any execution outside of itself.
If it failed, then may (likely) have callbacks executing after the text execution, but it's already a fail case, so later test-corruption isn't... any worse.
Certainly some tech debt for later, but for now, the standard/passing case is that it runs cleanly and correctly without any chance of cross-test-corruption in the passing case.